spec/controllers/fed_ax_api_controller_spec.rb

```diff
@@ -0,0 +1,240 @@
+require 'rails_helper'
+require "support/shared_controller_examples"
+
+RSpec.describe FedAxApiController, type: :controller do
+  # setting up some variables that should permit valid interactions
+  let(:valid_package) { { weight: 1.5, height: 1.5, width: 1.5, depth: 1.5, product_id: 1 } }
+  let(:another_valid_package) { { weight: 2.5, height: 2.5, width: 2.5, depth: 2.5, product_id: 2 } }
+  let(:valid_packages) { [valid_package, another_valid_package] }
+  let(:valid_origin) { { country: "US", state: "WA", city: "Seattle", zip: "98101" } }
+  let(:valid_destination) { { country: "US", state: "IL", city: "Chicago", zip: "60652", order_id: 3 } }
+  let(:shipping_choice_params) { { carrier: "UPS", service_type: "UPS Three-Day Select" } }
+
+  # setting up some variables that should break stuff
+  let(:invalid_package) { [{ weight: "apples", height: "bananas", width: "avocados", depth: "pineco" }] }
+  let(:invalid_origin) { { country: "US", state: "WA", city: "Houston", zip: "60652" } }
+  let(:invalid_destination) { { country: "US", state: "OH", city: "Boise", zip: "20002" } }
+
+  describe "GET #quote" do
+    context "valid inputs" do
+      before :each do
+        VCR.use_cassette("ShippingAPI_valid") do
+          get :quote, packages: valid_packages, origin: valid_origin, destination: valid_destination
+        end
+      end
+
+      it_behaves_like "HTTP 200 success"
+      it_behaves_like "HTTP JSON response"
+
+      context "the returned hash object" do
+        before :each do
+          @response = JSON.parse response.body
+        end
+
+        it "contains a successful status code" do
+          expect(@response["status"]).to eq 200
+        end
+
+        it_behaves_like "shipping quotes"
+      end
+    end
+
+    context "valid inputs with optional single service type parameter" do
+      before :each do
+        VCR.use_cassette("ShippingAPI_valid") do
+          get :quote, packages: valid_packages, origin: valid_origin, destination: valid_destination, shipping: shipping_choice_params
+        end
+      end
+
+      it_behaves_like "HTTP 200 success"
+      it_behaves_like "HTTP JSON response"
+
+      context "the returned json object" do
+        before :each do
+          @response = JSON.parse response.body
+        end
+
+        it "contains a successful status code" do
+          expect(@response["status"]).to eq 200
+        end
+
+        it "contains a quote hash" do
+          expect(@response["quote"].class).to eq Hash
+        end
+
+        context "the quote object" do
+          before :each do
+            @quote = @response["quote"]
+          end
+
+          it "contains the total price of shipping" do
+            expect(@quote["total_price"]).to_not be_nil
```

Comment:
If you've got a recorded vcr, don't you know the value of this so you can assert something more precise than not being nil.